tools/precommit.py

```diff
@@ -0,0 +1,141 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import sys
+import os
+from common_py import path
+from common_py.system.filesystem import FileSystem as fs
+from common_py.system.executor import Executor as ex
+from common_py.system.platform import Platform
+
+TESTS=['host', 'rpi2', 'nuttx', 'darwin', 'misc']
+BUILDTYPES=['debug', 'release']
+
+
+def parse_option():
+    parser = argparse.ArgumentParser(
+         description='IoT.js pre-commit script.',
+         epilog='If no arguments are given, runs full test.')
+    parser.add_argument('--test', choices=TESTS, action='append')
+    parser.add_argument('--buildtype', choices=BUILDTYPES, action='append')
+
+    option = parser.parse_args(sys.argv[1:])
+    if option.test is None:
+        option.test = TESTS
+    if option.buildtype is None:
+        option.buildtype = BUILDTYPES
+    return option
+
+
+def setup_nuttx_root(nuttx_root):
+    # Step 1
+    fs.maybe_make_directory(nuttx_root)
+    fs.chdir(nuttx_root)
+    if fs.exists('nuttx'):
+        fs.chdir('nuttx')
+        ex.check_run_cmd('git', ['pull'])
+        fs.chdir('..')
+    else:
+        ex.check_run_cmd('git', ['clone',
+                                 'https://bitbucket.org/nuttx/nuttx.git'])
+    if fs.exists('apps'):
+        fs.chdir('apps')
+        ex.check_run_cmd('git', ['pull'])
+        fs.chdir('..')
+    else:
+        ex.check_run_cmd('git', ['clone',
+                                 'https://bitbucket.org/nuttx/apps.git'])
+
+    # Step 2
+    fs.maybe_make_directory(fs.join(nuttx_root, 'apps', 'system', 'iotjs'))
+    for file in fs.listdir(fs.join(path.PROJECT_ROOT,
+                                   'targets','nuttx-stm32f4','app')):
+        fs.copy(fs.join(path.PROJECT_ROOT,'targets','nuttx-stm32f4','app',file),
+                fs.join(nuttx_root, 'apps', 'system', 'iotjs'))
+
+    # Step 3
+    fs.chdir(fs.join(nuttx_root, 'nuttx', 'tools'))
+    ex.check_run_cmd('./configure.sh', ['stm32f4discovery/usbnsh'])
+    fs.chdir('..')
+    fs.copy(fs.join(path.PROJECT_ROOT,
+                    'targets',
+                    'nuttx-stm32f4',
+                    'nuttx',
+                    '.config.default'),
+            fs.join('.config'))
+
+
+def build_nuttx(nuttx_root, buildtype):
+    fs.chdir(fs.join(nuttx_root, 'nuttx'))
+    try:
+        nuttx_bin = fs.join(nuttx_root, 'nuttx', 'nuttx.bin')
+        if fs.exists(nuttx_bin):
+            fs.remove(nuttx_bin)
+        if buildtype == "release":
+            ex.run_cmd('make', ['IOTJS_ROOT_DIR=' + path.PROJECT_ROOT, 'R=1'])
+        else:
+            ex.run_cmd('make', ['IOTJS_ROOT_DIR=' + path.PROJECT_ROOT, 'R=0'])
+
+        if fs.exists(nuttx_bin):
+            return True
+        else:
+            print "Cannot create nuttx.bin"
```

Comment:
How about chaning error messasge to `Failed to build nuttx` like others such as jerry, libtuv, or IoT.js